logic/transaction.js

```diff
@@ -65,1057 +51,1165 @@ __private.types = {};
- *
- * @see {@link privateTypes}
- * @param {transaction} transaction
- * @param {block} block
- * @param {account} sender
- * @param {function} cb - Callback function
- * @returns {SetImmediate} error
- * @todo Add description for the params
- */
-Transaction.prototype.apply = function(transaction, block, sender, cb, tx) {
-	if (!this.ready(transaction, sender)) {
-		return setImmediate(cb, 'Transaction is not ready');
-	}
+		// Check confirmed sender balance
+		let amount = new bignum(transaction.amount.toString()).plus(
+			transaction.fee.toString()
+		);
+		const senderBalance = this.checkBalance(
+			amount,
+			'balance',
+			transaction,
+			sender
+		);
 
-	// Check confirmed sender balance
-	var amount = new bignum(transaction.amount.toString()).plus(
-		transaction.fee.toString()
-	);
-	var senderBalance = this.checkBalance(amount, 'balance', transaction, sender);
+		if (senderBalance.exceeded) {
+			return setImmediate(cb, senderBalance.error);
+		}
 
-	if (senderBalance.exceeded) {
-		return setImmediate(cb, senderBalance.error);
-	}
+		amount = amount.toNumber();
 
-	amount = amount.toNumber();
-
-	this.scope.logger.trace('Logic/Transaction->apply', {
-		sender: sender.address,
-		balance: -amount,
-		blockId: block.id,
-		round: slots.calcRound(block.height),
-	});
-	this.scope.account.merge(
-		sender.address,
-		{
+		this.scope.logger.trace('Logic/Transaction->apply', {
+			sender: sender.address,
 			balance: -amount,
 			blockId: block.id,
 			round: slots.calcRound(block.height),
-		},
-		(err, sender) => {
-			if (err) {
-				return setImmediate(cb, err);
-			}
-			/**
-			 * calls apply for Transfer, Signature, Delegate, Vote, Multisignature,
-			 * DApp, InTransfer or OutTransfer.
-			 */
-			__private.types[transaction.type].apply.call(
-				this,
-				transaction,
-				block,
-				sender,
-				err => {
-					if (err) {
-						this.scope.account.merge(
-							sender.address,
-							{
-								balance: amount,
-								blockId: block.id,
-								round: slots.calcRound(block.height),
-							},
-							err => setImmediate(cb, err),
-							tx
-						);
-					} else {
-						return setImmediate(cb);
-					}
-				},
-				tx
-			);
-		},
-		tx
-	);
-};
+		});
+		this.scope.account.merge(
+			sender.address,
+			{
+				balance: -amount,
+				blockId: block.id,
+				round: slots.calcRound(block.height),
+			},
+			(err, sender) => {
+				if (err) {
+					return setImmediate(cb, err);
+				}
+				/**
+				 * calls apply for Transfer, Signature, Delegate, Vote, Multisignature,
+				 * DApp, InTransfer or OutTransfer.
+				 */
+				__private.types[transaction.type].apply.call(
+					this,
+					transaction,
+					block,
+					sender,
+					err => {
+						if (err) {
+							this.scope.account.merge(
+								sender.address,
+								{
+									balance: amount,
+									blockId: block.id,
+									round: slots.calcRound(block.height),
+								},
+								err => setImmediate(cb, err),
```

Comment:
We should rename all `err` errors to something else, so that they don't shadow existing errors.

Reply:
as we discussed we need to investigate this further, so I am not going to fix this as part of this PR.